### PR TITLE
BUG: GH35865 - int cast to str with colon setitem

### LINKED ISSUE #35865: BUG: ser[:] = ["a", 1] casts 1 to "1"
```
ser = pd.Series(["a", 1])
ser[:] = list(ser.values))

>>> ser[1]
'1'
```

Looks like we call `np.array(ser.values)` and the cast happens in there.

### PR DESCRIPTION
**Pull request by arredond**
- [x] closes #35865 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry: not sure if it's needed but happy to provide one given some guidelines

### CODE REVIEW

File: pandas/tests/indexing/test_coercion.py [hunk 1]
```diff
@@ -258,6 +258,13 @@ def test_setitem_series_timedelta64(self, val, exp_dtype):
         )
         self._assert_setitem_series_conversion(obj, val, exp, exp_dtype)
 
+    def test_setitem_series_colon(self):
+        # GH35865 - int casted to str when internally calling np.array(ser.values)
+        ser = pd.Series(["a", 1])
+        ser[:] = list(ser.values)
+
+        assert ser[1] == 1
```
Comment (phofl) [inline pandas/tests/indexing/test_coercion.py:266]: please use tm.assert_series_equal and check the whole object.

I think it is enough to link the issue and give the test a bit more of a specific name

**Comment by phofl** [comment]
Don't think we need a whatsnew.

File: pandas/tests/indexing/test_coercion.py (hunk 1 above)
Comment (arredond) [reply pandas/tests/indexing/test_coercion.py:266]: Do you mean something like this?

```python
def test_setitem_series_colon(self):
    # GH35865 - int casted to str when internally calling np.array(ser.values)
    ser1 = pd.Series(["a", 1])
    ser2 = pd.Series(["a", 1)]
    ser2[:] = list(ser2.values)

    tm.assert_series_equal(ser1, ser2)
```

Also, I'm unsure what an appropriate name would be. I was following the `test_setitem_series` structure from the file. Maybe `test_setitem_series_no_coercion_from_values_list`? Might be too long

Comment (arredond) [reply pandas/tests/indexing/test_coercion.py:266]: Also, what do you mean by linking the issue? The number is referenced in the first comment of the test, should I add the full url?

Comment (phofl) [reply pandas/tests/indexing/test_coercion.py:266]: Length is ok, name sounds good.

sorry no I meant only adding the gh reference, no additional comment

Comment (arredond) [reply pandas/tests/indexing/test_coercion.py]: Done!

**Review by phofl — APPROVED** [review]

File: pandas/tests/indexing/test_coercion.py [hunk 2]
```diff
@@ -258,6 +258,14 @@ def test_setitem_series_timedelta64(self, val, exp_dtype):
         )
         self._assert_setitem_series_conversion(obj, val, exp, exp_dtype)
 
+    def test_setitem_series_no_coercion_from_values_list(self):
+        # GH35865 - int casted to str when internally calling np.array(ser.values)
+        ser1 = pd.Series(["a", 1])
```
Comment (jreback) [inline pandas/tests/indexing/test_coercion.py:263]: can you call one of these 'exp' or 'expected'
then use call the other ser

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_coercion.py (hunk 2 above)
Comment (arredond) [reply pandas/tests/indexing/test_coercion.py]: Done!

**Comment by jreback** [comment]
thanks @arredond

**Pull request merged**